### ISSUE TITLE
iOS 10 Rich Push notifications

### ISSUE THREAD

**Issue by aurusapps** [williamrijksen/com.williamrijksen.onesignal#13]
Are iOS 10 Rich Push notifications ie. with buttons and media supported in the current version of the module?

**Comment by jkasten2** [comment]
@aurusapps Yes, this supported by this SDK. Make sure you have the latest version in your app. Also make sure your media links don't contain any redirects and are HTTPS. HTTP URLs can be used but require a plist entry to override the HTTPS requirement.

Also you may have to enable "Remote notifications" under Background Modes in Xcode if titanium isn't doing it for you.
https://documentation.onesignal.com/docs/ios-sdk-setup#section-2-add-required-capabilities

**Comment by aurusapps** [comment]
@jkasten2 My question was whether it is implemented in this module rather than the native iOS SDK itself. Would you know if it is? I feel the SDK in use is older and maybe we need to recompile this module with the latest version of the SDK.

**Comment by jkasten2** [comment]
@aurusapps Although it isn't documented in this repo I see it is using 2.1.16 of the OneSignal iOS native SDK. Support for iOS 10 media attachments was added back in version 2.1.1 of the native SDK so this plugin will support it was well.

The requirement of the 2.1.16 native SDK is that you must have `"Remote notifications"` under `Background Modes` enabled from our [native instructions](https://documentation.onesignal.com/docs/ios-sdk-setup#section-3-add-required-capabilities) (also noted above.) If you do not do this then the notification will not show at all if you add action buttons or an attachment to it.

There are a few more requirements on the iOS device itself but that is the only requirement for your app. I recommend following troubleshooting the guide below to check this on the device end.
https://documentation.onesignal.com/docs/troubleshooting-ios#section-notification-display-issues

If you are stuck let us know if you are not seeing the notification at all when you add an attachment or is the notification displaying but just without the attachment?

**Comment by aurusapps** [comment]
my bad, this seems to have worked on another app we tried it with. Will update with the exact steps we used so others can also use rich notifications in iOS.

**Comment by chmiiller** [comment]
Not sure, but I think latest commit also provides a way to use Rich Notifications on iOS. Can you confirm this @williamrijksen ? Than you may want to also close this issue =)

**Comment by jvandijk** [comment]
@chmiiller Theoretically yes. By providing the documented extension in step 5 of the [README](https://github.com/williamrijksen/com.williamrijksen.onesignal/blob/master/README.md) to your app, you should be able to use rich notifications. We haven't tested this though...

**Comment by chmiiller** [comment]
hey @jkasten2 do you mind having a look at #38 ? Maybe you can assist us on that? Thanks!

**Comment by jvandijk** [comment]
In #38 we've update to the latest version of OneSignal 2.5.4

We did not test the rich notifications though, because we haven't been using these ourselves.

**Comment by agrycroko** [comment]
Hi all, so as of now, January 20th 2018, what's the process for getting media links to show as little thumb images on iOS. I put a media url (https and ending with .png) into the OneSignal push field but only the message is displayed, not the thumb image. iOS 11.2 app in development mode on iPhone X. Ti SDK 7.0.1

**Comment by hansemannn** [comment]
Btw: This should be possible once [this PR](https://github.com/appcelerator/titanium_mobile/pull/8078) is merged. If you want to help getting it merged sooner than later, please try out the PR and provide feedback about possible issues.

**Comment by hansemannn** [comment]
Looks like this 😇:
<img width="300" alt="bildschirmfoto 2018-03-22 um 17 24 08" src="https://user-images.githubusercontent.com/10667698/37783479-e40607ce-2df5-11e8-826b-685af7f25c47.png"> <img width="300" alt="bildschirmfoto 2018-03-22 um 17 23 59" src="https://user-images.githubusercontent.com/10667698/37783480-e43b8106-2df5-11e8-81ba-1168b41e2f93.png">

**Comment by williamrijksen** [comment]
@hansemannn Did you test this in combination with the onesignal module?

**Comment by SilvioLuis** [comment]
@williamrijksen  Do we have any resolution on this topic? I could not get it to work, even following the steps to enable rich notifications on iOS.
